File: HRO/CMITDW Taal der Wiskunde/tentamen/Lua/o5.lua
local numbers = {4, 6, 8, 10, 12, 14}
local times = {}
local bingo = {} -- bingo card

local function wipe(t)
	for k in pairs(t) do
		t[k] = nil
	end
end

local function check(t)
	for _, v in ipairs(numbers) do
		if not t[v] then
			return false
		end
	end
	return true
end

local function average(t)
	local sum = 0
	for _, v in ipairs(t) do
		sum = sum + v
	end
	return sum / #t
end

for i = 1, 1e5 do -- number of times
	local tries = 0
	wipe(bingo)
	
	while (true) do
		tries = tries + 1
		local sum = 0
		
		for j = 1, 3 do
			sum = sum + math.random(6)
		end
		
		--print(sum)
		bingo[sum] = true
		
		if check(bingo) then
			times[i] = tries
			--print(i, tries)
			break
		end
	end
end

print(average(times))

File: edu/HRO/CMITDW Taal der Wiskunde/tentamen/o5.java
import java.util.Random;

public class o5
{
	public static int[] numbers = {4, 6, 8, 10, 12, 14};
	
	public static void main(String[] args)
	{
		Random rnd = new Random();
		int[] times = new int[100000]; // number of times
		boolean[] bingo = new boolean[18]; // bingo card
		
		for (int i=0; i<times.length; i++)
		{
			int tries = 0;
			clear(bingo);
			
			while (true)
			{
				tries++;
				int sum = 0;
				
				for (int j=1; j<=3; j++) // number of dices
					sum += (1 + rnd.nextInt(6));
				
				//System.out.println(sum);
				bingo[sum-1] = true;
				
				if (check(bingo))
				{
					times[i] = tries;
					//System.out.println(i+"\t"+tries);
					break;
				}
			}
		}
		
		System.out.println(average(times));
	}
	
	public static double average(int t[])
	{
		double sum = 0;
		for (int v : t)
			sum += v;
		return sum / t.length;
	}
	
	public static void clear(boolean t[])
	{
		for (int i=0; i<t.length; i++)
			t[i] = false;
	}
	
	public static boolean check(boolean t[])
	{
		for (int v : numbers)
		{
			if (!t[v-1])
				return false;
		}
		return true;
	}
}

